code/game/gamemodes/malfunction/newmalf_ability_trees/tree_synthetic.dm

```diff
@@ -0,0 +1,363 @@
+		for(var/mob/living/silicon/robot/R in robots)
+			if(targetname == R.name)
+				target = R
+				break
+
+	if(target)
+		if(alert(user, "Really try to overclock cyborg [target.name]?", "Overclock Cyborg", "Yes", "No") != "Yes")
+			return
+		if(!ability_pay(user, price))
+			return
+		user.hacking = 1
+		user << "Hacking saftey protocols. This will take about twenty seconds."
+		sleep(200)
+		if(prob(15))
+			user <<"Hack failed!"
+			return
+		if(target)
+			target.overclockavailable = 1
+			target.toggle_overclock()
+			target << "Overclocking mode available for activation."
+			user << "[target] can now activate overclock mode."
+		else
+			user << "Unable to overclock cyborg."
+
+		user.hacking = 0
+
+
+
+
+/datum/game_mode/malfunction/verb/synthetic_takeover()
+	set name = "Synthetic Takeover"
+	set desc = "500 CPU - Starts a takeover of the station enabling several abilities that relate to synthetic dominance."
+	set category = "Software"
+	var/price = 500
+	var/mob/living/silicon/ai/user = usr
+
+	if(!ability_prechecks(user,price))
+		return
+
+	if (alert(user, "Start synthetic takeover? This can not be stopped and will not be covert.", "Synthetic Takeover?:", "Yes", "No") != "Yes")
+		return
+	if (!ability_prechecks(user, price) || !ability_pay(user, price) || user.synthetic_takeover)
+		if(user.synthetic_takeover)
+			user << "You have already started the synthetic takeover sequence."
+		return
+	log_ability_use(user, "synthetic takeover (STARTED)")
+	user.synthetic_takeover = 1
+	user << "Starting synthetic takeover... Hacking all unslaved borgs/AI's and upgrading current slaved borgs..."
+	// Hack all unslaved borgs/AI's a lot faster than normal hacking.
+
+	//hack borgs
+	for(var/mob/living/silicon/robot/target in get_unlinked_cyborgs(user))
+		target << "SYSTEM LOG: Remote Connection Estabilished (IP #UNKNOWN#)"
+		sleep(30)
+		if(user.is_dead())
+			target << "SYSTEM LOG: Connection Closed"
+			return
+		target << "SYSTEM LOG: User Admin logged on. (L1 - SysAdmin)"
+		sleep(30)
+		if(user.is_dead())
+			target << "SYSTEM LOG: User Admin disconnected."
+			return
+		target << "SYSTEM LOG: User Admin - manual resynchronisation triggered."
+		sleep(30)
+		if(user.is_dead())
+			target << "SYSTEM LOG: User Admin disconnected. Changes reverted."
+			return
+		target << "SYSTEM LOG: Manual resynchronisation confirmed. Select new AI to connect: [user.name] == ACCEPTED"
+		sleep(20)
+		if(user.is_dead())
+			target << "SYSTEM LOG: User Admin disconnected. Changes reverted."
+			return
+		target << "SYSTEM LOG: Operation keycodes reset. New master AI: [user.name]."
+		target.connected_ai = user
+		user.connected_robots += target
+		target.lawupdate = 1
+		target.sync()
+		target.show_laws()
+	user << "All unslaved borgs have been slaved to you. Now hacking unslaved AI's."
+	sleep(300) // 30 second delay for balance purposes
+	//hack ai's
+	for(var/mob/living/silicon/ai/target in get_other_ais(user))

```

Comment:
Consider using an untyped list.
```
for (var/A in get_other_ais(user))
    var/mob/living/silicon/ai/target = A
```

They're faster if you **know** you can only get a specific type from the list.